### PR TITLE
Bug 1270432 – Add support for long press actions to menu items.

### PR DESCRIPTION
**Pull request by jhugman**
https://bugzilla.mozilla.org/show_bug.cgi?id=1270432

### CODE REVIEW

File: Client/Frontend/Widgets/Menu/MenuView.swift [hunk 1]
```diff
@@ -223,6 +225,19 @@ class MenuView: UIView {
         toolbarButtonSelected(view)
     }
 
+    @objc private func menuLongPressed(recognizer: UILongPressGestureRecognizer) {
+        guard recognizer.state == .Began else {
+            return
+        }
+        let collectionView: UICollectionView = self.menuPagingView
```
Comment (fluffyemily) [inline Client/Frontend/Widgets/Menu/MenuView.swift]: not sure what the point of assigning this to another variable is. Are you trying to make it explicit that the `menuPagingView` is a collectionView for clarity?

File: Client/Frontend/Widgets/Menu/MenuViewController.swift [hunk 2]
```diff
@@ -186,6 +186,20 @@ extension MenuViewController: MenuItemDelegate {
         performMenuAction(menuItem.action, withAnimation: animation, onView: menuItemCell.menuImageView)
     }
 
+    func menuView(menuView: MenuView, didLongPressItemAtIndexPath indexPath: NSIndexPath) {
```
Comment (fluffyemily) [inline Client/Frontend/Widgets/Menu/MenuViewController.swift]: There is a lot of duplicated code here between `didLongPressItemAtIndexPath` and `didSelectItemAtIndexPath` around the highlighting of images etc. Can we maybe factor out the shared code into a separate function?

File: Client/Frontend/Widgets/Menu/MenuView.swift [hunk 3]
```diff
@@ -211,10 +211,7 @@ class MenuView: UIView {
     }
 
     @objc private func toolbarButtonSelected(sender: UIView) {
-        guard let selectedView = (toolbarItems.filter { item in
-            return item.customView != nil && item.customView! == sender
-        }).first,
-            let selectedItemIndex = toolbarItems.indexOf(selectedView) else {
+        guard let selectedItemIndex = toolbarItems.indexOf({ $0.customView == sender }) else {
```
Comment (fluffyemily) [inline Client/Frontend/Widgets/Menu/MenuView.swift]: Hmm, much simpler. Must've been crazy when I wrote this!

File: Client/Frontend/Widgets/Menu/MenuViewController.swift (hunk 2 above)
Comment (jhugman) [inline Client/Frontend/Widgets/Menu/MenuViewController.swift]: I was worried about using `selectedIcon`, when perhaps `selectedIcon` may not be appropriate for the `secondaryAction`. I was most concerned about using animations.

Will refactor.

**Pull request merged**